doc/rst/language/spec/procedures.rst

```diff
@@ -667,14 +667,12 @@ argument expression is a tuple of the actual arguments.
       4.0
 
    defines a generic procedure called ``mywriteln`` that takes a
-   variable number of arguments of any type and then writes them out on
-   separate lines. The parameter
-   for-loop (:ref:`Parameter_For_Loops`) is unrolled by the
-   compiler so that ``i`` is a parameter, rather than a variable. This
-   needs to be a parameter for-loop because the expression ``x(i)`` will
-   have a different type on each iteration. The type of ``x`` can be
-   specified in the formal argument list to ensure that the actuals all
-   have the same type.
+   variable number of arguments of any type and then writes them out
+   on separate lines. The for-loop over the heterogeneous
+   tuple (:ref:`Iteration_over_Tuples`) is unrolled by the compiler so
+   that each iteration has its own copy of ``x`` with the appropriate
```

Comment:
Do we unroll all loops if a heterogeneous tuple is used? How do we know the tuple here is heterogeneous? 

I think the first question is more for me but for the second question we may need some clarification in the doc.

Reply:
We do unroll them (and have to since the index variable representing the tuple elements is potentially different for each element; or at least, we have to create a case per type in the tuple).  We tell whether the tuple is heterogeneous because Chapel's a statically typed language so we can see the types of the tuple's elements in a value situation and differentiate between `(int, real, int)` and `(int, int, int)` or `3*int` in a type situation.

Reply:
Right. But the first call `mywriteln("hi", "there")` will cause this to be a homogeneous tuple and that will lead that loop to be not unrolled?

I am only worried about the vague/potentially confusing allusion to loop unrolling here. Even if you agree, it can be a follow up PR that fixes this part.

Reply:
> But the first call mywriteln("hi", "there") will cause this to be a homogeneous tuple and that will lead that loop to be not unrolled?

Right.  The compiler converts varargs procedures into procedures that take a single tuple argument and then distinguishes between the homogeneous and heterogeneous cases.  Ah, I see now that the unrolling comment seems pretty specific to heterogeneous tuples as though that was the only case handled here.  OK, let me finesse this a bit.

Reply:
See what you think about https://github.com/chapel-lang/chapel/pull/14522/commits/be229156fb85dcaa0fbb50caac23a3d241dbfa15.  I think in the previous revision I was feeling too beholden too the original wording which didn't apply anymore since the param for loop was gone (but I tried to preserve the point for no good reason).

Reply:
Yeah, sorry for asking indirect questions. That was partly due to me trying to understand what's happening and giving your initial wording the benefit of the doubt. I think the current wording is much less confusing.